packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -119,18 +134,151 @@ function getResizeDirection(element: HTMLElement, direction: Direction) {
   return side;
 }
 
+function preventClick(event: MouseEvent) {
+  event.preventDefault();
+  event.stopImmediatePropagation();
+}
+
+/**
+ * Checker that returns a promise that resolves when the column virtualization
+ * is disabled.
+ */
+function useColumnVirtualizationDisabled(apiRef: React.MutableRefObject<GridPrivateApiPro>) {
+  const promise = React.useRef<ControllablePromise>();
+  const selector = () => gridVirtualizationColumnEnabledSelector(apiRef);
+  const value = useGridSelector(apiRef, selector);
+
+  React.useEffect(() => {
+    if (promise.current && value === false) {
+      promise.current.resolve();
+      promise.current = undefined;
+    }
+  });
+
+  const asyncCheck = () => {
+    if (!promise.current) {
+      if (selector() === false) {
+        return Promise.resolve();
+      }
+      promise.current = createControllablePromise();
+    }
+    return promise.current;
+  };
+
+  return asyncCheck;
+}
+
+/**
+ * Basic statistical outlier detection, checks if the value is `F * IQR` away from
+ * the Q1 and Q3 boundaries. IQR: interquartile range.
+ */
+function excludeOutliers(inputValues: number[], factor: number) {
+  if (inputValues.length < 4) {
+    return inputValues;
+  }
+
+  const values = inputValues.slice();
+  values.sort((a, b) => a - b);
+
+  const q1 = values[Math.floor(values.length * 0.25)];
+  const q3 = values[Math.floor(values.length * 0.75) - 1];
+  const iqr = q3 - q1;
+
+  // We make a small adjustment if `iqr < 5` for the cases where the IQR is
+  // very small (e.g. zero) due to very close by values in the input data.
+  // Otherwise, with an IQR of `0`, anything outside that would be considered
+  // an outlier, but it makes more sense visually to allow for this 5px variance
+  // rather than showing a cropped cell.
+  const deviation = iqr < 5 ? 5 : iqr * factor;
+
+  return values.filter((v) => v > q1 - deviation && v < q3 + deviation);
+}
+
+function extractColumnWidths(
+  apiRef: React.MutableRefObject<GridPrivateApiPro>,
+  options: AutosizeOptionsRequired,
+) {
+  type Result = Record<string, number>;
+
+  const headers = apiRef.current.columnHeadersContainerElementRef?.current!;
+  const container = apiRef.current.virtualScrollerRef?.current!;
+
+  const getHeader = (field: string) =>
+    headers.querySelector(`:scope > div > div > [data-field="${field}"][role="columnheader"]`);
+
+  const getCells = (field: string) =>
+    Array.from(
+      container.querySelectorAll(`:scope > div > div > div > [data-field="${field}"][role="cell"]`),
+    );
+
+  const widthByField = options.columns.reduce((result, column) => {
```

Comment:
Should we filter out the columns with `resizable: false` in the updated width calculation?
Current behavior prevents autosizing such columns by separator double-click but [allows](https://codesandbox.io/s/condescending-bhaskara-x4m2h8) to autosize with the API method and `autosizeOnMount` prop. Is that intended?

Reply:
Non-resizable columns now abort during the double-click event handler. This will still leave the possibility for library-users to autosize columns via the API that browser-users can't resize manually.

Reply:
I think that's an interesting discussion, tbh, I don't know the exact answer but we have different behaviors regarding these flags in `GridColDef` (e.g resizable, hidable, sortable, editable, etc)

_For example:_
With sorting, `apiRef` method `sortColumn` doesn't support sorting if `sortable: false`.
With column visibility, `column.hideable` is only honoured on the UI side, it's possible to change it from the API.
 
As a general rule of thumb, flags like resizable, hidable, sortable etc should only limit the functionality on the UI side. It should be possible to achieve a similar thing using the API as you did in this example.

Should we think about making this behavior consistent, maybe in v7 as it seems breaking?

i.e. `GridColDef.flag: false` -> not allowed from the UI but allowed using the `apiRef.current.specificAPIMethod`

CC @cherniavskii

Reply:
Yeah I think we should adopt the convention that flags affect the UI actions, not the API.

Reply:
> Yeah I think we should adopt the convention that flags affect the UI actions, not the API.

Separating UI actions from the API is tricky.
Take `filterable` for example. Currently, `filterable: false` impacts both UI and API - for example, the filtering model ignores columns that are not filterable: https://codesandbox.io/s/mystifying-kirch-klhz8j?file=/Demo.tsx
If we change the behavior to only impact the UI, how should the demo above behave? Should it show the applied filter in the UI? Or filter the data without any visual indication in the UI?
What do you think?

Reply:
Yes, it should show the filter applied in the UI, but the filter should not be modifiable by the end-user. The filter model is controlled by the library-users, if they didn't want any filter applied, they have the option to not add a filter model.

This is a breaking change but I think we should migrate to that convention at V7. It's going to be much simpler & intuitive for us & library-users that flags are defined that way.

Reply:
I think `filterable: false` only impacting the UI side of things _does_ make sense, AFAI understand this, the purpose of exposing the api is to empower the developers to achieve something that is not easily achievable by the end user, this filtering use case is a good example of that.
And for a benchmark, if I am not mistaken, **AgGrid** also behaves in a similar fashion.